From 11889bb69b874879487be27f7859a6c419fecd56 Mon Sep 17 00:00:00 2001
From: Arnaud Vrac <avrac@freebox.fr>
Date: Wed, 22 Nov 2023 19:44:00 +0100
Subject: [PATCH 02/94] desktop-shell: set proper curtain size when no output
 is created yet

In this case the curtain width and height would be calculated using
uninitialized values, triggering warnings in pixman calls.

Signed-off-by: Arnaud Vrac <avrac@freebox.fr>
(cherry picked from commit 03fa71f153502a4d107de01c06381ab8a91a478e)
Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 desktop-shell/shell.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/desktop-shell/shell.c b/desktop-shell/shell.c
index 8deac7f9cd..4b62ade1b9 100644
--- a/desktop-shell/shell.c
+++ b/desktop-shell/shell.c
@@ -3846,7 +3846,7 @@ shell_fade_create_view(struct desktop_shell *shell)
 	};
 	struct weston_curtain *curtain;
 	bool first = true;
-	int x1, y1, x2, y2;
+	int x1 = 0, y1 = 0, x2 = 0, y2 = 0;
 
 	wl_list_for_each(shell_output, &shell->output_list, link) {
 		struct weston_output *op = shell_output->output;
-- 
2.20.1

